core/commands/external.go

```diff
@@ -52,8 +52,16 @@ func ExternalBinary() *cmds.Command {
 			// setup env of child program
 			env := os.Environ()
 
-			nd, err := req.InvocContext().GetNode()
-			if err == nil {
+			// Get the node iff already defined.
+			if req.InvocContext().Online {
+				nd, err := req.InvocContext().GetNode()
```

Comment:
we can probably just skip this `GetNode` call entirely and just use the `req.InvocContext().Online` value directly below for setting the variable

Reply:
I wanted to ensure we actually *created* the node. `Online` means "node will be constructed in online mode" but I don't know if it actually *constructs* the node. Actually, I'm pretty sure it doesn't.